e2e/cypress/integration/enterprise/ldap/ldap_group_sync_spec.js

```diff
@@ -0,0 +1,343 @@
+// ***************************************************************
+
+// Group: @enterprise @ldap
+
+function setLDAPTestSettings(config) {
+    return {
+        siteName: config.TeamSettings.SiteName,
+        siteUrl: config.ServiceSettings.SiteURL,
+        teamName: '',
+        user: null,
+    };
+}
+
+// assumes the CYPRESS_* variables are set
+// assumes that E20 license is uploaded
+// for setup with AWS: Follow the instructions mentioned in the mattermost/platform-private/config/ldap-test-setup.txt file
+context('ldap', () => {
+    let testChannel;
+    let testTeam;
+    let testUser;
+
+    describe('LDAP Group Sync Automated Tests', () => {
+        beforeEach(() => {
+            // * Check if server has license for LDAP
+            cy.apiRequireLicenseForFeature('LDAP');
+
+            // # Initial api setup
+            cy.apiInitSetup().then(({team, user}) => {
+                testTeam = team;
+                testUser = user;
+
+                // # Update LDAP settings
+                cy.apiGetConfig().then(({config}) => {
+                    setLDAPTestSettings(config);
+                });
+
+                // # Login as sysadmin and add board-one to test team
+                cy.apiAdminLogin();
+
+                // # Link board group
+                cy.visit('/admin_console/user_management/groups');
+                cy.get('#board_group').then((el) => {
+                    if (!el.text().includes('Edit')) {
+                        // # Link the Group if its not linked before
+                        if (el.find('.icon.fa-unlink').length > 0) {
+                            el.find('.icon.fa-unlink').click();
+                        }
+                    }
+                });
+
+                // # Link developers group
+                cy.visit('/admin_console/user_management/groups');
+                cy.get('#developers_group').then((el) => {
+                    if (!el.text().includes('Edit')) {
+                        // # Link the Group if its not linked before
+                        if (el.find('.icon.fa-unlink').length > 0) {
+                            el.find('.icon.fa-unlink').click();
+                        }
+                    }
+                });
+
+                // # Create a test channel
+                cy.apiCreateChannel(testTeam.id, 'ldap-group-sync-automated-tests', 'ldap-group-sync-automated-tests').then((response) => {
+                    testChannel = response.body;
+                });
+            });
+        });
+
+        it('MM-T1537 - Sync Group Removal from Channel Configuration Page', () => {
+            // # Login as sysadmin and link 2 groups to testChannel
+            cy.apiAdminLogin();
+            cy.visit(`/admin_console/user_management/channels/${testChannel.id}`);
+            cy.wait(2000); //eslint-disable-line cypress/no-unnecessary-waiting
+
+            // # Link first group
+            cy.findByTestId('addGroupsToChannelToggle').click();
```

Comment:
Prefer element timeout over explicit wait to avoid unnecessary waiting, especially to make sure that page is loaded after visiting a url before proceeding, e.g.

```suggestion
            cy.get('.admin-console__header', {timeout: TIMEOUTS.HALF_MIN}).should('be.visible').and('have.text', 'Channel Configuration');
            cy.findByTestId('addGroupsToChannelToggle').click();
```

Reply:
Same goes for the other `cy.wait()` calls below

Reply:
Hey! This became more challenging than I expected 😅 . I literally spent the entire day haha.
I was able to remove a lot of the cy.waits with this method however it proved rather difficult in some areas! Those areas were when the component had loaded but some of the data had not. I could not figure out a way to save my life on how to stop cypress from going forward without the cy.wait. 
Hope thats okay? I mean I tried my best to remove all the cy.wait's but some places I could not so I just left them in. 
Let me know what you think! Thanks!

Reply:
@hahmadia Yeah, that's ok. We just wanted to minimize the usage of cy.wait in the codebase. But if it's necessary, then we should be fine. Thanks for looking into!